### PR TITLE
Fix slurm script tests

### PR DESCRIPTION
**Pull request by mr0re1**
* Remove global `compute` that was slowing down tests;
* Remove unused optional args;
* Fix "NPE" in `create_client_options`;
* Refactor `test_create_client_options`.

### CODE REVIEW

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py [hunk 1]
```diff
@@ -83,10 +83,9 @@
 )
 
 
-def start_instance_op(inst, project=None):
-    project = project or lkp.project
-    return compute.instances().start(
-        project=project,
```
Comment (cdunbar13) [inline community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py:89]: Why remove the additional parameter?  Is it not possible to create VMs in a separate project from the controller?

(This question applies to the other locations where you removed project as a parameter)

**Review by cdunbar13 — COMMENTED** [review]
There are a few questions I'd like addressed in the comments.  Additionally, was there testing done to make sure the change from compute as a global to part of lkp didn't break things?

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py [hunk 2]
```diff
@@ -205,17 +203,17 @@ def get_credentials() -> Optional[service_account.Credentials]:
...
-def create_client_options(api: ApiEndpoint = None) -> ClientOptions:
+def create_client_options(api: Optional[ApiEndpoint] = None) -> ClientOptions:
     """Create client options for cloud endpoints"""
     ver = endpoint_version(api)
     ud = universe_domain()
     options = {}
     if ud and ud != DEFAULT_UNIVERSE_DOMAIN:
         options["universe_domain"] = ud
-    if ver:
+    if api and ver:
```
[the remark this answers is not in the document]
Comment (mr0re1) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py:213]: "ver is None if api is None" - this is implementation detail of `endpoint_version`.

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py [hunk 3]
```diff
@@ -205,17 +203,17 @@ def get_credentials() -> Optional[service_account.Credentials]:
     return credentials
 
 
-def create_client_options(api: ApiEndpoint = None) -> ClientOptions:
+def create_client_options(api: Optional[ApiEndpoint] = None) -> ClientOptions:
```
[the remark this answers is not in the document]
Comment (mr0re1) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py:206]: None of type annotations are necessary. But without `Optional` typechecker will not work as well.

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py [hunk 4]
```diff
@@ -1092,7 +1088,7 @@ def batch_callback(rid, resp, exc):
                 done[rid] = resp
 
     def batch_request(reqs):
-        batch = compute.new_batch_http_request(callback=batch_callback)
+        batch = lkp.compute.new_batch_http_request(callback=batch_callback)
```
[the remark this answers is not in the document]
Comment (mr0re1) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py:1091]: `lkp.compute` is cached.

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py (hunk 1 above)
Comment (mr0re1) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py:89]: I removed all optional that never used. This is technical debt.

**Review by cdunbar13 — APPROVED** [review]

**Pull request merged**